pkg/etcdcli/etcdcli.go

```diff
@@ -274,6 +275,20 @@ func (g *etcdClientGetter) GetMember(name string) (*etcdserverpb.Member, error)
 	return nil, apierrors.NewNotFound(schema.GroupResource{Group: "etcd.operator.openshift.io", Resource: "etcdmembers"}, name)
 }
 
+// If the member's name is not set, extract ip/hostname from peerURL. Useful with unstarted members.
+func GetMemberNameOrHost(member *etcdserverpb.Member) string {
+	memberName := member.Name
+	if len(memberName) == 0 {
+		u, err := url.Parse(member.PeerURLs[0])
+		if err != nil {
+			panic(fmt.Sprintf("Unstarted member has invalid peerURL: %v", err))
```

Comment:
errors in go don't generally start with capitals can you adjust please? Did you validate this justifies panic I commented it would be very bad but also a panic is very extreme? Just making sure your confidence in the situation.

Reply:
I think it would be pretty unusual to have peerURL to be empty. But panic would be extreme. We could print an error, and return a string `NO-NAME-BAD-PEER-URL` instead.

Reply:
Pushed a new commit removing the panic.